File: AS241S3_T15-FundoSmart/src/app/core/services/fruit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Fruit } from '../interfaces/fruit';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class FruitService {
private apiUrl = `${environment.apiUrl}/fruits`;
  constructor(private http: HttpClient) {}

  findAll(): Observable<Fruit[]> {
    return this.http.get<Fruit[]>(this.apiUrl);
  }

  findById(id: number): Observable<Fruit> {
    return this.http.get<Fruit>(`${this.apiUrl}/${id}`);
  }

  save(fruit: Fruit): Observable<Fruit> {
    return this.http.post<Fruit>(this.apiUrl, fruit);
  }

  update(fruit: Fruit): Observable<Fruit> {
    return this.http.put<Fruit>(this.apiUrl, fruit);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: gam/AS241S3_T15-be-feature-sp3_u3_market_customer 2/src/main/java/pe/edu/vallegrande/project/service/FruitService.java
package pe.edu.vallegrande.project.service;

import pe.edu.vallegrande.project.model.Fruit;
import java.util.List;
import java.util.Optional;

public interface FruitService {
    List<Fruit> findAll();
    Optional<Fruit> findById(Long id);
    Fruit save(Fruit fruit);
    Fruit update(Fruit fruit);
    void delete(Fruit fruit);
}
